### PR TITLE
Enhancements to databases documentation

### PR DESCRIPTION
**Pull request by goetzk**
I've tried to consolidate the permissions requirements (as they're identical on
all databases) and to improve readability added some formatting to the mysql
entry.
Example commands are added for postgresql, though the section is by no means a
full guide.

### CODE REVIEW

File: guides/InstallationAdminGuide.md [hunk 1]
```diff
@@ -58,12 +58,13 @@ openEQUELLA installation involves the following steps:
 
 All databases must be able to store character data using UTF-8 encoding.
 
-A database must be configured for use by openEQUELLA with openEQUELLA being the owner or having complete control of the database.
+A database must be configured for use by openEQUELLA with openEQUELLA being the owner or having complete control of the database. That means the equella database user must have permission to create, modify and delete tables, indexes and constraints and to run, select, insert, delete and update queries.
```
Comment (cbeach47) [inline guides/InstallationAdminGuide.md]: I would clarify `...and to run, select, insert, delete and update queries. `

to be something like the following.  as-is, makes it sound like 'run' is a query type: `...and to run the CRUD queries of: select, insert, delete and update. `

File: guides/InstallationAdminGuide.md [hunk 2]
```diff
@@ -75,22 +76,29 @@ openEQUELLA uses the default Users Tablespace that must have an <unlimited> Quot
 ### Create a Microsoft SQL Server database instance for openEQUELLA
 
 If you are using Microsoft SQL Server 2008, 2012 or 2014, please ensure that the TCP/IP protocol has been enabled. The openEQUELLA user (the installer default value is ‘equellauser’) must have the following:
-* Permissions: to create, modify and delete tables, indexes and constraints and to run, select, insert, delete and update queries. 
 * Database Name: the installer default name is ‘equella’.
 * Database Role: the database user must be the database owner. For the openEQUELLA user login select ‘db_owner’.
 
 It is required that Microsoft SQL Server databases have READ_COMMITTED_SNAPSHOT enabled to avoid possible deadlocks. See http://msdn.microsoft.com/en-us/library/ms173763.aspx for more information. The following statement will enable this setting for a given database:
 
+```
```
Comment (cbeach47) [inline guides/InstallationAdminGuide.md]: Add code block type:
```` ```sql ````

File: guides/InstallationAdminGuide.md [hunk 3]
```diff
@@ -75,22 +76,29 @@ openEQUELLA uses the default Users Tablespace that must have an <unlimited> Quot
...
 
 ### Create a PostgreSQL database instance for openEQUELLA
 
 Installations using a PostgreSQL (8.0 or higher) database must create an openEQUELLA database before installing openEQUELLA. The openEQUELLA user (in the default install this is ‘equellauser’) must have the following:
-* Permissions: to create, modify and delete tables, indexes and constraints and to run, select, insert, delete and update queries. 
 * Database Name: the default name used is ‘equella’.
 * Database Owner: the default name used is ‘equellauser’.
 * Tablespace:‘pg_default’.
 
+These requirements are met by the following commands
```
Comment (cbeach47) [inline guides/InstallationAdminGuide.md]: :+1: 
Consider adding:  From a terminal (not via psql)

Also, I needed to run as postgres to make the createuser / createdb commands work.
```bash
sudo -u postgres -i
```

File: guides/InstallationAdminGuide.md [hunk 4]
```diff
@@ -75,22 +76,29 @@ openEQUELLA uses the default Users Tablespace that must have an <unlimited> Quot
...
+
+```
+createuser --interactive --pwprompt equellauser
+createdb -O equellauser -D pg_default -E UTF-8 equella
+```
+
```
Comment (cbeach47) [inline guides/InstallationAdminGuide.md:104]: Consider adding:  For ```createuser```, you can respond 'n' for:
- Shall the new role be a superuser? (y/n) 
- Shall the new role be allowed to create databases? (y/n) 
- Shall the new role be allowed to create more new roles? (y/n)

File: guides/InstallationAdminGuide.md [hunk 5]
```diff
@@ -75,22 +76,29 @@ openEQUELLA uses the default Users Tablespace that must have an <unlimited> Quot
...
 
 Installations using a PostgreSQL (8.0 or higher) database must create an openEQUELLA database before installing openEQUELLA. The openEQUELLA user (in the default install this is ‘equellauser’) must have the following:
-* Permissions: to create, modify and delete tables, indexes and constraints and to run, select, insert, delete and update queries. 
 * Database Name: the default name used is ‘equella’.
 * Database Owner: the default name used is ‘equellauser’.
 * Tablespace:‘pg_default’.
 
+These requirements are met by the following commands
+
+```
```
Comment (cbeach47) [inline guides/InstallationAdminGuide.md]: Add code block type:
```` ```bash ````

**Review by cbeach47 — CHANGES_REQUESTED** [review]
Thanks for the consolidation and details!

I suggested some minor changes to include before we have this go live.

**Comment by goetzk** [comment]
Hi @cbeach47 ,
I've made changes based on your feedback. I haven't directly copied your suggestions but the forced reviewing meant I saw some other ways to phrase things.

File: guides/InstallationAdminGuide.md [hunk 6]
```diff
@@ -58,12 +58,16 @@ openEQUELLA installation involves the following steps:
 
 All databases must be able to store character data using UTF-8 encoding.
 
-A database must be configured for use by openEQUELLA with openEQUELLA being the owner or having complete control of the database.
+A database must be configured for use by openEQUELLA with openEQUELLA being the owner or having complete control of the database. That means the equella database user must have permission to:
+* Create, modify and delete tables, indexes and constraints
+* Perform select, insert, delete and update queries on the newly created databases.
+
+You must also ensure the database user has a password as openEQUELLA will request id during installation.
```
Comment (cbeach47) [inline guides/InstallationAdminGuide.md]: minor change -
> You must also ensure the database user has a password as openEQUELLA will request **id** during installation.

to
> You must also ensure the database user has a password as openEQUELLA will request **it** during installation.

File: guides/InstallationAdminGuide.md [hunk 7]
```diff
@@ -58,12 +58,16 @@ openEQUELLA installation involves the following steps:
 
 All databases must be able to store character data using UTF-8 encoding.
 
-A database must be configured for use by openEQUELLA with openEQUELLA being the owner or having complete control of the database.
+A database must be configured for use by openEQUELLA with openEQUELLA being the owner or having complete control of the database. That means the equella database user must have permission to:
```
Comment (cbeach47) [inline guides/InstallationAdminGuide.md]: > That means the **equella** database user must have permission to:

to
> That means the **openequella** database user must have permission to:

File: guides/InstallationAdminGuide.md (hunk 4 above)
Comment (cbeach47) [reply guides/InstallationAdminGuide.md:104]: The user doesn't need to be able to create databases nor be a superuser.

**Review by cbeach47 — CHANGES_REQUESTED** [review]
minor edits requested.

File: guides/InstallationAdminGuide.md (hunk 5 above)
Comment (goetzk) [reply guides/InstallationAdminGuide.md]: I used 'sh', hopefully highlights the same but happy to change to 'bash' should it be required.

File: guides/InstallationAdminGuide.md (hunk 7 above)
Comment (goetzk) [reply guides/InstallationAdminGuide.md]: I have this change staged locally

File: guides/InstallationAdminGuide.md (hunk 6 above)
Comment (goetzk) [reply guides/InstallationAdminGuide.md]: i have this change staged locally

**Comment by goetzk** [comment]
@cbeach47 Hopefully now in a suitable state.

**Review by cbeach47 — APPROVED** [review]

**Comment by cbeach47** [comment]
:+1:  looks good - thanks @goetzk !

**Pull request merged**